### PR TITLE
requireCapitalizedConstructorsNew: New rule based on jshint newcap.

### LINKED ISSUE #2046: jscs "requireCapitalizedConstructors" and jshint "newcap" are different
Basically `requireCapitalizedConstructors` works by requiring constructors after the `new` keyword to be capitalized, where as `newcap` works by requiring the `new` keyword before capitalized function calls.

Here is a test case to show the difference:

`.jscsrc`
```json
{
	"requireCapitalizedConstructors": true
}
```

`.jshintrc`
```json
{
	"newcap": true
}
```

`test.js`
```js
function Test() {
	return null;
}
function test() {
	return null;
}

var testUpper = Test();
var testLower = new test();
```

**jscs output:**
```
$ jscs .
Constructor functions should be capitalized at ./test.js :
     7 |
     8 |var testUpper = Test();
     9 |var testLower = new test();
----------------------------^
    10 |


1 code style error found.
```

**jshint output:**
```
$ jshint .
test.js: line 8, col 17, Missing 'new' prefix when invoking a constructor.

1 error
```

Personally, I would like to have both in jscs. Is there an option I'm missing to enforce this, or could one be added?

### PR DESCRIPTION
**Pull request by AlexanderOMara**
Added a new rule that behaves closer to the behavior of "newcap" in jshint.

Fixes #2046

### CODE REVIEW

File: test/specs/rules/require-capitalized-constructors-new.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+var Checker = require('../../../lib/checker');
+var expect = require('chai').expect;
+
+describe('rules/require-capitalized-constructors-new', function() {
+    var checker;
+
+    function baseCases() {
```
Comment (markelog) [inline test/specs/rules/require-capitalized-constructors-new.js]: Not sure if we need that, if tests covering used code-paths we good either way

Comment (markelog) [inline test/specs/rules/require-capitalized-constructors-new.js]: Means, that we could move those tests from the helper function and to the `'with `true` value'` `describe`

Comment (AlexanderOMara) [inline test/specs/rules/require-capitalized-constructors-new.js]: So it's not necessary to run these tests on both the `true` and `allExcept` configurations? I based it on [test/specs/rules/require-capitalized-constructors.js](https://github.com/jscs-dev/node-jscs/blob/master/test/specs/rules/require-capitalized-constructors.js)

Comment (markelog) [inline test/specs/rules/require-capitalized-constructors-new.js]: I'd say no, for my taste, if we covered it, it should be good

**Pull request closed without merging**